app/controllers/repositories_controller.rb

```diff
@@ -182,4 +225,24 @@ def create_equipments
         Equipment.create(name: e, repository_id: @repository.id)
       end if params['equipments'].present?
     end
+
+    def update_share_type
```

Comment:
yes because to update the password, my conditions require the share_type to be saved. Also, for the "if @repository.update(repository_params)" to pass, we need the password to be in place. otherwise it will be false and will got to else statement. I figured this was the only way that everything could work. To save the share_type in order to update the password, then the if statement passes correctly. Hope this made sense and if there is a better way to approach this, please let me know.